### PR TITLE
test: test deployment for bytecode variance independence

### PR DESCRIPTION
**Pull request by cucupac**

### CODE REVIEW

File: test/create3/Create3Factory.t.sol [hunk 1]
```diff
@@ -129,13 +132,38 @@ contract Create3FactoryTest is DeploymentHelper, TestSetup {
         assertEq(setOneChildB, setTwoChildB, "Equivalence violation: setOneChildB != setTwoChildB");
     }
 
+    function testFuzz_DeployBytecodeVarianceIndependence(uint256 pkNum) public {
+        // Setup
+        VmSafe.Wallet memory wallet = vm.createWallet(uint256(keccak256(abi.encodePacked(uint256(pkNum)))));
+        bytes memory constructorArgsBytecode1 = abi.encode(wallet.addr);
+        bytes memory constructorArgsBytecode2 = abi.encode(CONTRACT_DEPLOYER);
+
```
Comment (rahul0eth) [inline test/create3/Create3Factory.t.sol]: Add:
`assert(constructorArgsBytecode1 != constructorArgsBytecode2, "Equivalence check: constructorArgsBytecode1 == constructorArgsBytecode2");`

File: test/create3/Create3Factory.t.sol [hunk 2]
```diff
@@ -118,8 +121,8 @@ contract Create3FactoryTest is DeploymentHelper, TestSetup {
         vm.revertTo(snapShot);
 
         // Second deployment set (reverse order)
-        deployChild(address(proxy), wallet, strippedBytecodeChildB);
-        deployChild(address(proxy), wallet, strippedBytecodeChildA);
+        deployChild(address(proxy), wallet, strippedBytecodeChildB, "");
```
Comment (0xernesto) [inline test/create3/Create3Factory.t.sol]: Shouldn't this be "0x"?

File: test/create3/Create3Factory.t.sol [hunk 3]
```diff
@@ -105,11 +107,12 @@ contract Create3FactoryTest is DeploymentHelper, TestSetup {
...
         // Setup
         VmSafe.Wallet memory wallet = vm.createWallet(uint256(keccak256(abi.encodePacked(uint256(pkNum)))));
+        bytes memory constructorArgsBytecode = abi.encode(wallet.addr);
 
         // First deployment set
         uint256 snapShot = vm.snapshot();
-        deployChild(address(proxy), wallet, strippedBytecodeChildA);
-        deployChild(address(proxy), wallet, strippedBytecodeChildB);
+        deployChild(address(proxy), wallet, strippedBytecodeChildA, constructorArgsBytecode);
+        deployChild(address(proxy), wallet, strippedBytecodeChildB, "");
```
Comment (0xernesto) [inline test/create3/Create3Factory.t.sol]: Shouldn't this be "0x"?

**Review by rahul0eth — CHANGES_REQUESTED** [review]
See comment.

**Review by 0xernesto — CHANGES_REQUESTED** [review]

File: test/create3/Create3Factory.t.sol (hunk 2 above)
Comment (cucupac) [reply test/create3/Create3Factory.t.sol]: According to [this](https://ethereum.stackexchange.com/questions/6618/in-solidity-how-can-you-set-a-bytes-variable-to-empty), no.

File: test/create3/Create3Factory.t.sol (hunk 3 above)
Comment (cucupac) [reply test/create3/Create3Factory.t.sol]: According to [this](https://ethereum.stackexchange.com/questions/6618/in-solidity-how-can-you-set-a-bytes-variable-to-empty), no.

**Review by 0xernesto — APPROVED** [review]

File: src/create3/Create3Factory.sol [hunk 4]
```diff
@@ -114,18 +116,18 @@ contract Create3Factory is Initializable, UUPSUpgradeable, Ownable {
...
      */
     function deploy(
         address _principal,
         bytes memory _signature,
-        bytes memory _strippedBytecode,
+        bytes memory _creationCode,
         bytes memory _constructorArgsBytecode
     ) public payable {
-        bytes32 hashedStrippedBytecode = keccak256(_strippedBytecode);
+        bytes32 hashedStrippedBytecode = keccak256(_creationCode);
```
Comment (0xernesto) [inline src/create3/Create3Factory.sol]: change to hashedCreationCode

File: src/create3/interfaces/ICreate3Factory.sol [hunk 5]
```diff
@@ -3,14 +3,14 @@ pragma solidity ^0.8.21;
...
+    function getTransactionHash(address _principal, bytes memory _creationCode) external returns (bytes32);
+    function getAddress(address _principal, bytes memory _creationCode) external returns (address);
+    function getBytecodeHash(bytes memory _creationCode) external pure returns (bytes32);
     function getDeploymentHistory(address _principal) external returns (address[] memory);
     function deploy(
         address _principal,
         bytes memory _signature,
-        bytes memory _strippedBytecode,
+        bytes memory _creationCode,
         bytes memory _constructorArgsBytecode
```
Comment (0xernesto) [inline src/create3/interfaces/ICreate3Factory.sol]: change to _constructorArgsCode

File: src/create3/Create3Factory.sol [hunk 6]
```diff
@@ -114,18 +116,18 @@ contract Create3Factory is Initializable, UUPSUpgradeable, Ownable {
...
-     * @param _strippedBytecode The bytecode of the contract to be deployed without the constructor arguments.
+     * @param _creationCode The bytecode of the contract to be deployed without the constructor arguments.
      * @param _constructorArgsBytecode The encoded constructor arguments of the contract to be deployed.
      */
     function deploy(
         address _principal,
         bytes memory _signature,
-        bytes memory _strippedBytecode,
+        bytes memory _creationCode,
         bytes memory _constructorArgsBytecode
```
Comment (0xernesto) [inline src/create3/Create3Factory.sol]: change all instance of _constructorArgsBytecode to _constructorArgsCode

File: test/create2/common/contracts/Child.args.t.sol [hunk 7]
```diff
@@ -0,0 +1,13 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.21;
+
+import { IChild } from "../interfaces/IChild.t.sol";
+
+// Value types
```
Comment (0xernesto) [inline test/create2/common/contracts/Child.args.t.sol]: Remove comment

File: test/create3/helpers/DeploymentHelper.t.sol [hunk 8]
```diff
@@ -3,21 +3,54 @@ pragma solidity ^0.8.21;
...
-     * @param _strippedBytecode The to-be-deployed contract bytecode.
+     * @param _creationCode The to-be-deployed contract bytecode.
      */
-    function deployChild(address _contract, VmSafe.Wallet memory _wallet, bytes memory _strippedBytecode) internal {
-        bytes32 txHash = ICreate3Factory(_contract).getTransactionHash(_wallet.addr, _strippedBytecode);
+    function deployChild(
+        address _contract,
+        VmSafe.Wallet memory _wallet,
+        bytes memory _creationCode,
+        bytes memory _constructorArgsBytecode
```
Comment (0xernesto) [inline test/create3/helpers/DeploymentHelper.t.sol]: change _constructorArgsBytecode to _constructorArgsCode

**Review by 0xernesto — CHANGES_REQUESTED** [review]

**Review by rahul0eth — CHANGES_REQUESTED** [review]
See comments.

**Review by rahul0eth — APPROVED** [review]

**Review by 0xernesto — APPROVED** [review]

**Review by rahul0eth — APPROVED** [review]

**Pull request merged**